projects/plugins/jetpack/modules/search/class-jetpack-search-customberg.php

```diff
@@ -0,0 +1,145 @@
+<?php
+/**
+ * A class that adds a search customization interface to wp-admin.
+ *
+ * @package automattic/jetpack
+ */
+
+namespace Automattic\Jetpack\Search;
+
+use Jetpack_Plan;
+use Jetpack_Search_Helpers;
+
+require_once JETPACK__PLUGIN_DIR . '_inc/lib/admin-pages/class.jetpack-admin-page.php';
+require_once JETPACK__PLUGIN_DIR . '_inc/lib/admin-pages/class-jetpack-redux-state-helper.php';
+
+/**
+ * Responsible for adding a search customization interface to wp-admin.
+ *
+ * @package Automattic\Jetpack\Search
+ */
+class Jetpack_Search_Customberg {
+	/**
+	 * The singleton instance of this class.
+	 *
+	 * @var Jetpack_Search_Customberg
+	 */
+	protected static $instance;
+
+	/**
+	 * Get the singleton instance of the class.
+	 *
+	 * @return Jetpack_Search_Customberg
+	 */
+	public static function instance() {
+		if ( ! isset( self::$instance ) ) {
+			self::$instance = new Jetpack_Search_Customberg();
+			self::$instance->init_hooks();
+		}
+
+		return self::$instance;
+	}
+
+	/**
+	 * Adds action hooks.
+	 */
+	public function init_hooks() {
+		add_action( 'admin_menu', array( $this, 'add_wp_admin_page' ), 999 );
+	}
+
+	/**
+	 * Adds a wp-admin page without adding a sidebar submenu item.
+	 */
+	public function add_wp_admin_page() {
+		if ( ! $this->should_add_page() ) {
+			return;
+		}
+
+		// Intentionally omits adding a submenu via the first null argument.
+		$hook = add_submenu_page(
+			null,
+			__( 'Search Settings', 'jetpack' ),
+			__( 'Search', 'jetpack' ),
+			'manage_options',
+			'jetpack-search-customize',
+			array( $this, 'jetpack_search_admin_page' )
+		);
+		add_action( "admin_print_scripts-$hook", array( $this, 'load_admin_scripts' ) );
+		add_action( "admin_print_styles-$hook", array( $this, 'load_admin_styles' ) );
+	}
+
+	/**
+	 * Prints the dashboard container.
+	 *
+	 * @access public
+	 */
+	public function jetpack_search_admin_page() {
+		// TODO: Spin this function off into a static helper function in a helper class for code reuse.
+		$protocol   = is_ssl() ? 'https' : 'http';
```

Comment:
Part of me wonders if we should worry about http. We used to in the past, but now that HTTPS is widespread I think we could opt to always serve https in these cases. What do you think?

Reply:
Just realized we could omit this altogether and use a `//en.wordpress.com/i/loading/loading-64.gif` here, no need to manually specify the protocol!